frontend/components/Comment.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react'
+import Link from 'next/link'
+import Form from '../components/Form'
+import { Consumer } from '../Context'
+class Comment extends Component {
+  state = { editValue: '', errors: [], textfieldOpen: false }
+  change = event => {
+    this.setState({ editValue: event.target.value })
+  }
+  componentDidMount() {
+    this.setState(prevState => ({
+      editValue: this.props.content
+    }))
+  }
+  render() {
+    return (
+      <React.Fragment>
+        <div>
+          {this.props.content}
+          <br />
+          {this.props.authorName}
+          <br />
+          {this.props.createdAt}{' '}
+          {this.props.canEditOrDelete ? (
+            <React.Fragment>
+              <span
+                onClick={() => {
+                  this.setState(prevState => ({
+                    textfieldOpen: !prevState.textfieldOpen
+                  }))
+                }}
+              >
+                {this.state.textfieldOpen ? 'Cancel' : 'Edit'}
+              </span>
+              |<span onClick={this.props.deleteComment}>Delete</span>
+              {this.state.textfieldOpen ? (
+                <Form
+                  submit={async () => {
+                    await this.props.editComment(this.state.editValue)
```

Comment:
You don't have to keep passing `editComment` function down as props since you made it globally accessible from using context and logically its just a function with no real dependencies on any other component's data. I can see it requires postID which you can easily get from the parent or using `next-router` since im guessing the route looks something likes this`/post/postID`

Reply:
I passed it down through props so we can modify the callback based on the page. Right now editComment in context just edits the comment, while editComment in [id].js shows a message and updates state.